### PR TITLE
feat: experimental React Suspense support in useSession hook

### PR DESCRIPTION
**Pull request by cesarenaldi**

### CODE REVIEW

File: packages/react/src/profile/__tests__/useProfileActionHistory.spec.ts [hunk 1]
```diff
@@ -8,14 +8,13 @@ import { waitFor } from '@testing-library/react';
 import { setupHookTestScenarioWithSession } from '../../__helpers__/setupHookTestScenarioWithSession';
 import { useProfileActionHistory } from '../useProfileActionHistory';
 
-// TODO make it work
-describe.skip(`Given the ${useProfileActionHistory.name} hook`, () => {
```
Comment (krzysu) [inline packages/react/src/profile/__tests__/useProfileActionHistory.spec.ts:12]: thanks for fixing it, I remember I couldn't figure out why it didn't want to work

File: packages/react/src/authentication/useSession.ts [hunk 2]
```diff
@@ -121,38 +141,167 @@ export type Session = AnonymousSession | ProfileSession | WalletOnlySession;
...
+
+  // Handle suspense
+  if (args?.suspense) {
+    if (result.loading) {
+      // Do the same thing as the effect right above because the effect won't run
+      // when we suspend.
+
+      throw new Promise<void>((resolve) => {
+        sessionDataVar.onNextChange(async (newSessionData) => {
+          await update(newSessionData); // TODO serve????
```
Comment (krzysu) [inline packages/react/src/authentication/useSession.ts:224]: I see that throwing a promise is a react suspense thing, anything else you wanted to add with your TODO comment?

**Review by krzysu — APPROVED** [review]
LGTM!

File: packages/react/src/authentication/useSession.ts (hunk 2 above)
Comment (cesarenaldi) [reply packages/react/src/authentication/useSession.ts:224]: no, I forgot to remove the TODO.

**Pull request merged**